core/src/execution/datafusion/expressions/regexp.rs

```diff
@@ -0,0 +1,144 @@
+        write!(
+            f,
+            "RLike [child: {}, pattern: {}] ",
+            self.child, self.pattern
+        )
+    }
+}
+
+impl PartialEq<dyn Any> for RLike {
+    fn eq(&self, other: &dyn Any) -> bool {
+        down_cast_any_ref(other)
+            .downcast_ref::<Self>()
+            .map(|x| self.child.eq(&x.child) && self.pattern.eq(&x.pattern))
+            .unwrap_or(false)
+    }
+}
+
+impl PhysicalExpr for RLike {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn data_type(&self, _input_schema: &Schema) -> datafusion_common::Result<DataType> {
+        Ok(DataType::Boolean)
+    }
+
+    fn nullable(&self, input_schema: &Schema) -> datafusion_common::Result<bool> {
+        self.child.nullable(input_schema)
+    }
+
+    fn evaluate(&self, batch: &RecordBatch) -> datafusion_common::Result<ColumnarValue> {
+        if let ColumnarValue::Array(v) = self.child.evaluate(batch)? {
+            if let ColumnarValue::Scalar(ScalarValue::Utf8(Some(pattern))) =
+                self.pattern.evaluate(batch)?
+            {
+                // TODO cache Regex across invocations of evaluate() or create it in constructor
+                match Regex::new(&pattern) {
+                    Ok(re) => {
+                        let inputs = v
+                            .as_any()
+                            .downcast_ref::<StringArray>()
+                            .expect("string array");
+                        let mut builder = BooleanBuilder::with_capacity(inputs.len());
+                        if inputs.is_nullable() {
+                            for i in 0..inputs.len() {
+                                if inputs.is_null(i) {
+                                    builder.append_null();
+                                } else {
+                                    builder.append_value(re.is_match(inputs.value(i)));
+                                }
+                            }
+                        } else {
+                            for i in 0..inputs.len() {
+                                builder.append_value(re.is_match(inputs.value(i)));
+                            }
+                        }
+                        Ok(ColumnarValue::Array(Arc::new(builder.finish())))
+                    }
+                    Err(e) => Err(CometError::Internal(format!(
+                        "Failed to compile regular expression: {e:?}"
+                    ))
+                    .into()),
+                }
+            } else {
+                Err(
+                    CometError::Internal("Only scalar regex patterns are supported".to_string())
+                        .into(),
+                )
+            }
+        } else {
+            // this should be unreachable because Spark will evaluate regex expressions against
+            // literal strings as part of query planning
+            Err(CometError::Internal("Only columnar inputs are supported".to_string()).into())
```

Comment:
Is it possible to encounter dictionary type?

Reply:
Good point. Yes, it probably is.